### PR TITLE
feat: add RDS primary key transformer

### PR DESCRIPTION
**Pull request by phani-srikar**
#### Description of changes

- Update the primary key index transformer to add support for generating appropriate AppSync pipeline resolvers for RDS imported data source.
- Update the schema generation to handle partition and sort key fields in the primary key.

#### Description of how you validated changes
Expanded the current unit tests to be run for RDS primary key transformer.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-index-transformer/src/graphql-index-transformer.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ import {
   ObjectTypeDefinitionNode,
 } from 'graphql';
 import { isListType, isScalarOrEnum } from 'graphql-transformer-common';
-import { appendSecondaryIndex, constructSyncVTL, updateResolversForIndex, getResourceOverrides, getDeltaSyncTableTtl } from './resolvers';
+import { appendSecondaryIndex, constructSyncVTL, updateResolversForIndex, getResourceOverrides, getDeltaSyncTableTtl } from './resolvers/resolvers';
```
Comment (marcvberg) [inline packages/amplify-graphql-index-transformer/src/graphql-index-transformer.ts:22]: You should still be able to do imports without the lower resolution as long as the right lines are added to the index.ts there

File: packages/amplify-graphql-index-transformer/src/graphql-primary-key-transformer.ts [hunk 2]
```diff
@@ -18,27 +19,35 @@ import {
...
   updateMutationConditionInput,
+  createHashField,
+  ensureModelSortDirectionEnum,
+  tryAndCreateSortField
 } from './schema';
 import { PrimaryKeyDirectiveConfiguration } from './types';
 import {
   validateNotSelfReferencing,
   validateNotOwnerAuth,
+  lookupResolverName
```
Comment (marcvberg) [inline packages/amplify-graphql-index-transformer/src/graphql-primary-key-transformer.ts:48]: nit: Linter probably wants a trailing comma here

**Review by marcvberg — APPROVED** [review]
Minor nits, overall LGTM

File: packages/amplify-graphql-index-transformer/src/resolvers/generators/rds-vtl-generator.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    if (updateResolver) {
+      addIndexToResolverSlot(updateResolver, [primaryKeySnippet]);
+    }
+
+    if (deleteResolver) {
+      addIndexToResolverSlot(deleteResolver, [primaryKeySnippet]);
+    }
+  };
```
Comment (sundersc) [inline packages/amplify-graphql-index-transformer/src/resolvers/generators/rds-vtl-generator.ts:62]: Do we need these for RDS?

File: packages/amplify-graphql-index-transformer/src/resolvers/generators/rds-vtl-generator.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  addIndexToResolverSlot,
+  getResolverObject,
+  mergeInputsAndDefaultsSnippet,
+  validateSortDirectionInput
+} from '../resolvers';
+import {
+  PrimaryKeyVTLGenerator,
+} from "./vtl-generator";
+
+export class RDSPrimaryKeyVTLGenerator implements PrimaryKeyVTLGenerator {
```
Comment (sundersc) [inline packages/amplify-graphql-index-transformer/src/resolvers/generators/rds-vtl-generator.ts:25]: Can we name the classes as '*IndexVTLGenerator' to be in sync with package name? We would use these classes for @index as well.

File: packages/amplify-graphql-index-transformer/src/resolvers/generators/vtl-generator.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import {
+  TransformerContextProvider,
+  TransformerResolverProvider
+} from '@aws-amplify/graphql-transformer-interfaces';
+
+import { PrimaryKeyDirectiveConfiguration } from '../../types';
+
+export interface PrimaryKeyVTLGenerator {
+  generate(config: PrimaryKeyDirectiveConfiguration, ctx: TransformerContextProvider, resolverMap: Map<TransformerResolverProvider, string>): void;
```
Comment (sundersc) [inline packages/amplify-graphql-index-transformer/src/resolvers/generators/vtl-generator.ts:9]: Index directive lives in the same package, so we need to name the method more specific to primaryKey.

File: packages/amplify-graphql-index-transformer/src/resolvers/generators/rds-vtl-generator.ts (hunk 3 above)
Comment (phani-srikar) [reply packages/amplify-graphql-index-transformer/src/resolvers/generators/rds-vtl-generator.ts:62]: Yes we do, in order to set the `WHERE` conditions based on the columns used in primary key.

File: packages/amplify-graphql-index-transformer/src/resolvers/generators/vtl-generator.ts (hunk 5 above)
Comment (phani-srikar) [reply packages/amplify-graphql-index-transformer/src/resolvers/generators/vtl-generator.ts:9]: Update to `generatePrimaryKeyInfo`

**Review by sundersc — APPROVED** [review]

**Pull request merged**